globset/src/glob.rs

```diff
@@ -194,6 +194,8 @@ struct GlobOptions {
     /// Whether to require a literal separator to match a separator in a file
     /// path. e.g., when enabled, `*` won't match `/`.
     literal_separator: bool,
+    /// doc
+    from_regex: bool,
```

Comment:
> I'd rather see Glob itself be an enum

Could you provide a draft definition for that enum that will suit the project best?
I've tried to replace the Glob structure with the
```
pub enum Pattern {
    Glob { glob: String, re: String, opts: GlobOptions, tokens: Tokens },
    Regex(String),
}
```
but it seems that enums don't have constructors in rust so the private GlobOptions and Tokens types leak.